tests/operator_test.go

```diff
@@ -503,6 +504,145 @@ var _ = Describe("[rfe_id:4784][crit:high] CDI Operator deployment + CDI CR dele
+					"watch",
+				},
+			},
+		}
+
+		_, err = f.K8sClient.RbacV1().ClusterRoles().Update(context.TODO(), clusterRole, metav1.UpdateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func() bool {
+			role, err := f.K8sClient.RbacV1().ClusterRoles().Get(context.TODO(), "cdi.kubevirt.io:config-reader", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			By("Waiting until list verb exists")
+			for _, verb := range role.Rules[0].Verbs {
+				if verb == "list" {
+					return true
+				}
+			}
+			return false
+		}, 2*time.Minute, 1*time.Second).Should(BeTrue())
+	})
+
+	It("ServiceAccount secrets restored on deletion attempt", func() {
+		serviceAccount, err := f.K8sClient.CoreV1().ServiceAccounts(f.CdiInstallNs).Get(context.TODO(), common.ControllerServiceAccountName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Remove secrets from ServiceAccount")
+		serviceAccount.Secrets = []corev1.ObjectReference{}
+
+		_, err = f.K8sClient.CoreV1().ServiceAccounts(f.CdiInstallNs).Update(context.TODO(), serviceAccount, metav1.UpdateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func() bool {
+			sa, err := f.K8sClient.CoreV1().ServiceAccounts(f.CdiInstallNs).Get(context.TODO(), common.ControllerServiceAccountName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			By("Waiting until secrets are repopulated")
+			return len(sa.Secrets) != 0
+		}, 2*time.Minute, 1*time.Second).Should(BeTrue())
+	})
+
+	It("Certificate restored to ConfigMap on deletion attempt", func() {
+		configMap, err := f.K8sClient.CoreV1().ConfigMaps(f.CdiInstallNs).Get(context.TODO(), "cdi-apiserver-signer-bundle", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Empty ConfigMap's data")
+		configMap.Data = map[string]string{}
+
+		_, err = f.K8sClient.CoreV1().ConfigMaps(f.CdiInstallNs).Update(context.TODO(), configMap, metav1.UpdateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func() bool {
+			cm, err := f.K8sClient.CoreV1().ConfigMaps(f.CdiInstallNs).Get(context.TODO(), "cdi-apiserver-signer-bundle", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			By("Waiting until ConfigMap's data is not empty")
+			return len(cm.Data) != 0
+		}, 2*time.Minute, 1*time.Second).Should(BeTrue())
+	})
+
+	It("Cant enable featureGate by editing CDIConfig resource", func() {
+		feature := "nonExistantFeature"
+		cdiConfig, err := f.CdiClient.CdiV1beta1().CDIConfigs().Get(context.TODO(), common.ConfigName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Enable non existant featureGate")
+		cdiConfig.Spec = cdiv1.CDIConfigSpec{
+			FeatureGates: []string{feature},
+		}
+
+		_, err = f.CdiClient.CdiV1beta1().CDIConfigs().Update(context.TODO(), cdiConfig, metav1.UpdateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func() bool {
+			config, err := f.CdiClient.CdiV1beta1().CDIConfigs().Get(context.TODO(), common.ConfigName, metav1.GetOptions{})
```

Comment:
I have a suspicion that the failure we are seeing, is because Update + Get might mean we get the old pre-update CDIConfig.

So we finish the test before the CDIConfig is reconciled to a normal value, and a few later tests fail.

Reply:
One way we can handle this is to check that the metadata.generation increased, and I'm surprised no test currently does this.